### ISSUE TITLE
Support Inheritance of OneToMany, ManyToOne fields, Kundera 2.8

### ISSUE THREAD

**Issue by atlanticdecisionsciences** [Impetus/kundera#439]
I have a class, DefaultInventoryLocation, annotated with @Inheritance:

@Entity
@Table(name = "InventoryLocation", schema = AtlanticKunderaTest@Atlantic_Mongo_PU_Test")
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "location_class")
public class DefaultInventoryLocation { ....

@ManyToOne(cascade = { CascadeType.ALL }, fetch = FetchType.EAGER)
@JoinColumn(name="LOCATION_TYPE_ID")
private LocationType locationType;

Class Tank extends it:

@Entity
@Table(name = "InventoryLocation", schema = "AtlanticKunderaTest@Atlantic_Mongo_PU_Test")
@DiscriminatorValue(value = "Tank")
 public class Tank extends DefaultInventoryLocation {...

If I create a new DefaultInventoryLocation and persist it, Kundera persists the associated LocationType as expected.  If I create a new Tank and persist it, the associated LocationType is not persisted.

**Comment by mevivs** [comment]
It is already supported with 2.8 release.

-Vivek

**Comment by atlanticdecisionsciences** [comment]
I attempted it last night, and it failed.  I'll try to modify one of your unit tests to illustrate.

**Comment by atlanticdecisionsciences** [comment]
Made the following changes to your Mongo unit tests.  The new OneToMany TransactionCatagory table is created in Mongo, but the entity is not persisted.

Transaction class:
- \* Copyright 2013 Impetus Infotech.
  package com.impetus.kundera.client.crud.mappedsuperclass;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.DiscriminatorColumn;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

/**
- @author vivek.mishra Transaction concrete entity
- */
  @Entity
  @Table(name = "TRNX")
  @Inheritance(strategy = InheritanceType.SINGLE_TABLE)
  @DiscriminatorColumn(name = "tx_type")
  public class Transaction extends AbstractTransaction {
  
  @Column
  private String bankIdentifier;
  
  public String getBankIdentifier() {
      return bankIdentifier;
  }
  
  public void setBankIdentifier(String bankIdentifier) {
      this.bankIdentifier = bankIdentifier;
  }
  
  @ManyToOne(cascade = { CascadeType.ALL }, fetch = FetchType.EAGER)
  @JoinColumn(name = "Transaction_ID")
  private TransactionCatagory catagory;
  
  public TransactionCatagory getCatagory() {
      return catagory;
  }
  
  public void setCatagory(TransactionCatagory catagory) {
      this.catagory = catagory;
  }

New class TransactionCatagory:

package com.impetus.kundera.client.crud.mappedsuperclass;

import javax.persistence.Column;

@Entity
@Table(name = "TransactionCatagory")
public class TransactionCatagory {

```
@Id
@GeneratedValue(strategy = GenerationType.AUTO)
private String _id;


@Column(name = "categoryName")
private String categoryName;

public String getCategoryName() {
    return categoryName;
}

public void setCategoryName(String categoryName) {
    this.categoryName = categoryName;
}

public String get_id() {
    return _id;
}

public void set_id(String _id) {
    this._id = _id;
}
```

}

Change to unit test:

public abstract class MappedSuperClassBase
{

```
/** log for this class. */
private static Logger log = LoggerFactory.getLogger(MappedSuperClassBase.class);

protected String _PU = "corePu";

/** The emf. */
protected static EntityManagerFactory emf;

/** The em. */
protected static EntityManager em;

protected void setUpInternal() throws Exception
{
    emf = Persistence.createEntityManagerFactory(_PU);
    em = emf.createEntityManager();
}

protected void assertInternal()
{
    assertInternal(false);
}

protected void assertInternal(boolean wait)
{
    CreditTransaction creditTx = new CreditTransaction();
    creditTx.setTxId("credit1");
    creditTx.setTxStatus(Status.APPROVED);
    creditTx.setBankIdentifier("sbi");
    creditTx.setTransactionDt(new Date());
    creditTx.setAmount(10);

    //  Add a many-to-one entity 
    TransactionCatagory cat = new TransactionCatagory();
    cat.setCategoryName("Name");
    creditTx.setCatagory(cat);


    em.persist(creditTx);

    waitThread(wait);

    DebitTransaction debitTx = new DebitTransaction();
    debitTx.setTxId("debit1");
    debitTx.setTxStatus(Status.PENDING);
    debitTx.setTransactionDt(new Date());
    debitTx.setBankIdentifier("sbi");
    debitTx.setAmount(-10);
    em.persist(debitTx);

    waitThread(wait);
    em.clear();
    String creditQuery = "Select c from CreditTransaction c where c.bankIdentifier = 'sbi'";

    Query query = em.createQuery(creditQuery);

    List<CreditTransaction> results = query.getResultList();
    Assert.assertEquals(1, results.size());
    Assert.assertEquals("credit1", results.get(0).getTxId());

    em.clear();
    String debitQuery = "Select d from DebitTransaction d where d.bankIdentifier = 'sbi'";

    query = em.createQuery(debitQuery);

    List<DebitTransaction> debitResults = query.getResultList();
    Assert.assertEquals(1, debitResults.size());
    Assert.assertEquals("debit1", debitResults.get(0).getTxId());


    em.clear();
    String oneToManyQuery = "Select e from TransactionCatagory e where e.categoryName = 'Name'";

    query = em.createQuery(oneToManyQuery);

    List<TransactionCatagory> entityResults = query.getResultList();
    Assert.assertEquals(1, entityResults.size());
    Assert.assertEquals("Name", entityResults.get(0).getCategoryName());

}

/**
 * Tear down.
 * 
 * @throws Exception
 *             the exception
 */
protected void tearDownInternal() throws Exception
{
    if (emf != null)
    {
        emf.close();
    }

    if (em != null)
    {
        em.close();
    }

    KunderaMetadata.INSTANCE.setApplicationMetadata(null);
}

private void waitThread(boolean toWait)
{
    if (toWait)
    {
        try
        {
            Thread.sleep(2000);
        }
        catch (InterruptedException e)
        {
            log.error("Error while thread interruption, {}", e);
        }
    }
}
```

}

**Comment by kkmishra** [comment]
Hi,
I confirm it is a bug and will be fixed in 2.9 release.
As a work around you can move relations into separate entity.
In your case if you move following code

```
@ManyToOne(cascade = { CascadeType.ALL }, fetch = FetchType.EAGER)
@JoinColumn(name = "Transaction_ID")
private TransactionCatagory catagory;
```

in CreditTransaction.java enity then it will work for you.

Thanks
Kuldeep
